From cd8976459d1c2ccf17a55c3cb78c2f5ea97b5529 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Fri, 13 Mar 2015 14:54:46 +0800
Subject: [PATCH 331/338] atomisp2: Revise zoom factor configuration

This patch revises the configuration of zoom factor
by setting the zoom factors to atomisp_css_params
instead of directly setting to atomisp_css_isp_config.

Change-Id: I3c9726457a2501c02a6544d3f53e7b15ae3d930f
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Fixes-Issue: #H1503936301
Reviewed-on: https://icggerrit.ir.intel.com/46221
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Build: IMIN-legacy (SC)
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Jing Tao <jing.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.c  | 17 +++++++++--------
 .../media/pci/atomisp2/atomisp_driver/atomisp_fops.c    | 10 +++++++---
 2 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 9ac7d2a..3303e00 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -3841,19 +3841,20 @@ void atomisp_css_set_zoom_factor(struct atomisp_sub_device *asd,
 {
 	struct atomisp_device *isp = asd->isp;
 
-	if (!asd->params.config.dz_config)
-		asd->params.config.dz_config = &asd->params.dz_config;
-
-	if (zoom == asd->params.config.dz_config->dx &&
-		 zoom == asd->params.config.dz_config->dy) {
+	if (zoom == asd->params.css_param.dz_config.dx &&
+		 zoom == asd->params.css_param.dz_config.dy) {
 		dev_dbg(isp->dev, "same zoom scale. skipped.\n");
 		return;
 	}
 
-	memset(asd->params.config.dz_config, 0,
+	memset(&asd->params.css_param.dz_config, 0,
 		sizeof(struct ia_css_dz_config));
-	asd->params.dz_config.dx = zoom;
-	asd->params.dz_config.dy = zoom;
+	asd->params.css_param.dz_config.dx = zoom;
+	asd->params.css_param.dz_config.dy = zoom;
+
+	asd->params.css_param.update_flag.dz_config =
+		(struct atomisp_dz_config *) &asd->params.css_param.dz_config;
+	asd->params.css_update_params_needed = true;
 }
 
 void atomisp_css_set_formats_config(struct atomisp_sub_device *asd,
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
index d714961..2f195c7 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
@@ -275,9 +275,13 @@ int atomisp_q_video_buffers_to_css(struct atomisp_sub_device *asd,
 			if (param->params.update_flag.dz_config &&
 				asd->run_mode->val != ATOMISP_RUN_MODE_VIDEO
 				&& !err) {
-				atomisp_css_set_dz_config(asd,
-						&param->params.dz_config);
-				atomisp_css_update_isp_params(asd);
+				memcpy(&asd->params.css_param.dz_config,
+					&param->params.dz_config,
+					sizeof(struct ia_css_dz_config));
+				asd->params.css_param.update_flag.dz_config =
+					(struct atomisp_dz_config *)
+					&asd->params.css_param.dz_config;
+				asd->params.css_update_params_needed = true;
 			}
 			/* free the parameters */
 			atomisp_free_css_parameters(&param->params);
-- 
1.9.1

